staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/utils.go

```diff
@@ -203,31 +203,83 @@ func VerifyVolumePathsForVMDevices(vmDevices object.VirtualDeviceList, volPaths
 
 // isvCenterDeprecated takes vCenter version and vCenter API version as input and return true if vCenter is deprecated
 func isvCenterDeprecated(vCenterVersion string, vCenerAPIVersion string) (bool, error) {
-	minvcversion, err := semver.New(MinvCenterVersion)
-	vcdeprecated := false
+	var vcversion, vcapiversion, minvcversion VCVersion
+	var err error
+	err = vcversion.New(vCenterVersion)
 	if err != nil {
-		return false, fmt.Errorf("failed to get parse vCenter version: %s. err: %+v", MinvCenterVersion, err)
-	} else {
-		vcversion, err := semver.New(vCenterVersion)
+		return false, fmt.Errorf("failed to get parse vCenter version: %s. err: %+v", vCenterVersion, err)
+	}
+	err = vcapiversion.New(vCenerAPIVersion)
+	if err != nil {
+		return false, fmt.Errorf("failed to get parse vCenter API version: %s. err: %+v", vCenerAPIVersion, err)
+	}
+	err = minvcversion.New(MinvCenterVersion)
+	if err != nil {
+		return false, fmt.Errorf("failed to get minimum vCenter version: %s. err: %+v", MinvCenterVersion, err)
+	}
+	if vcversion.Compare(minvcversion) == -1 || vcapiversion.Compare(minvcversion) == -1 {
+		return true, nil
+	}
+	return false, nil
+}
+
+// VCVersion represents a VC version
+type VCVersion struct {
+	Major    int64
+	Minor    int64
+	Revision int64
+	Build    int64
+}
+
+// New helps parse version string to VCVersion
+// returns error when parse fail
+func (v *VCVersion) New(version string) error {
+	for index, value := range strings.Split(version, ".") {
+		var err error
+		if index == 0 {
+			v.Major, err = strconv.ParseInt(value, 10, 64)
+		} else if index == 1 {
+			v.Minor, err = strconv.ParseInt(value, 10, 64)
+		} else if index == 2 {
+			v.Revision, err = strconv.ParseInt(value, 10, 64)
+		} else if index == 3 {
+			v.Build, err = strconv.ParseInt(value, 10, 64)
+		}
 		if err != nil {
-			return false, fmt.Errorf("failed to parse vCenter version: %s. err: %+v", vCenterVersion, err)
-		} else {
-			result := vcversion.Compare(*minvcversion)
-			if result == -1 {
-				// vcversion is less than minvcversion
-				vcdeprecated = true
-			} else if result == 0 {
-				// vcversion is equal to minvcversion
-				// check patch version
-				vcapiversion, err := semver.ParseTolerant(vCenerAPIVersion)
-				if err != nil {
-					return false, fmt.Errorf("failed to parse vCenter api version: %s. err: %+v", vCenerAPIVersion, err)
-				}
-				if vcapiversion.Patch < 3 {
-					vcdeprecated = true
-				}
-			}
+			return fmt.Errorf("failed to parse version: %q, err: %v", version, err)
+		}
+	}
+	return nil
+}
+
+// Compare compares VCVersion v to o:
+// -1 == v is less than o
+// 0 == v is equal to o
+// 1 == v is greater than o
+func (v *VCVersion) Compare(o VCVersion) int {
```

Comment:
`IsLessThan` would be probably better name (+ return bool) - nobody is interested in "equal" or "greater than" results. And why is it global?

Reply:
done.

```golang
// isLessThan compares VCVersion v to o and returns
// true if v is less than o
func (v *vcVersion) isLessThan(o vcVersion) bool {
```